core/common/src/unsafe/UnsafeBufferOperations.kt

```diff
@@ -85,6 +90,10 @@ public object UnsafeBufferOperations {
         buffer: Buffer,
         readAction: (bytes: ByteArray, startIndexInclusive: Int, endIndexExclusive: Int) -> Int
     ) {
+        contract {
```

Comment:
@JakeWharton could you please also plug the same contracts into extension functions declared in `UnsafeBufferOpertationsJvm.kt`?

Reply:
Yep!

Since these surely will conflict with #364, I'll wait for that to land then rebase & update this one.

Reply:
Rebased and updated.